### PR TITLE
sl-43: My Projects

### PR DESCRIPTION
**Pull request by vbazhynov**
image
image

### CODE REVIEW

File: frontend/src/pages/my-projects/my-projects.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// import { useEffect } from 'react';
+// import { useDispatch } from 'react-redux';
```
Comment (kuslosolodka) [inline frontend/src/pages/my-projects/my-projects.tsx]: B20

File: frontend/src/slices/projects/projects.slice.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import { DataStatus } from '~/libs/enums/enums.js';
+import { type ValueOf } from '~/libs/types/types.js';
+import { type ProjectGetAllResponseDto } from '~/packages/projects/projects.js';
+
+import { getProjects } from './actions.js';
+
+type State = {
+  dataStatus: ValueOf<typeof DataStatus>;
+  projects: ProjectGetAllResponseDto | null;
```
Comment (kuslosolodka) [inline frontend/src/slices/projects/projects.slice.ts:11]: `projects: ProjectGetAllResponseDto[]`

File: frontend/src/slices/projects/projects.slice.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+};
+
+const initialState: State = {
+  dataStatus: DataStatus.IDLE,
+  projects: null,
```
Comment (kuslosolodka) [inline frontend/src/slices/projects/projects.slice.ts:16]: projects: []

File: frontend/src/slices/projects/projects.slice.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  initialState,
+  name: 'projects',
+  reducers: {},
+  extraReducers(builder) {
+    builder.addCase(getProjects.pending, (state) => {
+      state.dataStatus = DataStatus.PENDING;
+    });
+    builder.addCase(getProjects.fulfilled, (state, action) => {
+      state.dataStatus = DataStatus.FULFILLED;
+      state.projects = action.payload;
```
Comment (kuslosolodka) [inline frontend/src/slices/projects/projects.slice.ts:29]: `state.projects = action.payload.items`

File: backend/src/packages/projects/project.service.ts [hunk 5]
```diff
@@ -23,6 +25,24 @@ class ProjectService implements Omit<IService, 'find' | 'update' | 'delete'> {
     };
   }
 
+  public async findByUserId({
+    userId,
+  }: FindProjectByUserIdRequest): Promise<ProjectGetAllResponseDto> {
+    // const items = await this.projectRepository.findByUserId(id);
```
Comment (kuslosolodka) [inline backend/src/packages/projects/project.service.ts:31]: B20

File: frontend/src/slices/projects/projects.slice.ts (hunk 2 above)
Comment (kuslosolodka) [inline frontend/src/slices/projects/projects.slice.ts]:
```suggestion
  projects: ProjectGetAllItemResponseDto[];
```

File: frontend/src/slices/projects/projects.slice.ts (hunk 3 above)
Comment (kuslosolodka) [inline frontend/src/slices/projects/projects.slice.ts]:
```suggestion
  projects: [],
```

File: frontend/src/slices/projects/projects.slice.ts (hunk 4 above)
Comment (kuslosolodka) [inline frontend/src/slices/projects/projects.slice.ts]:
```suggestion
      state.projects = action.payload.items;
```

File: backend/src/packages/projects/project.service.ts (hunk 5 above)
Comment (kuslosolodka) [inline backend/src/packages/projects/project.service.ts]: B20

File: backend/src/packages/projects/libs/types/types.ts [hunk 6]
```diff
@@ -1,5 +1,7 @@
+type FindProjectByUserIdRequest = { userId: string };
```
Comment (kuslosolodka) [inline backend/src/packages/projects/libs/types/types.ts]: it shouldn't be a dto?

File: backend/src/packages/projects/project.controller.ts [hunk 7]
```diff
@@ -46,6 +48,15 @@ class ProjectController extends Controller {
       handler: () => this.findAll(),
     });
 
+    this.addRoute({
+      path: ProjectsApiPath.USER_ID,
+      method: 'GET',
+      handler: (options) =>
+        this.findByUserId(
+          options as ApiHandlerOptions<{ params: FindProjectByUserIdRequest }>,
```
Comment (dimapopovych) [inline backend/src/packages/projects/project.controller.ts]:
```suggestion
          options as ApiHandlerOptions<{ user: UserAuthResponse }>,
```
And route probably should be projects/

File: backend/src/packages/projects/project.service.ts [hunk 8]
```diff
@@ -23,6 +25,24 @@ class ProjectService implements Omit<IService, 'find' | 'update' | 'delete'> {
...
+
+    const items = await this.projectRepository.findByUserId(userId);
+
+    if (!items) {
+      throw new ApplicationError({
+        message: 'This User Has No Projects',
```
Comment (dimapopovych) [inline backend/src/packages/projects/project.service.ts]: I think you should just return empty array in this case

File: frontend/src/index.tsx [hunk 9]
```diff
@@ -40,6 +42,10 @@ createRoot(document.querySelector('#root') as HTMLElement).render(
                 path: AppRoute.SIGN_UP,
                 element: <Auth />,
               },
+              {
+                path: AppRoute.MY_PROJECTS,
+                element: <MyProjects />,
```
Comment (dimapopovych) [inline frontend/src/index.tsx]: ProtectedRoute

File: frontend/src/libs/components/image/image.tsx [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+type Properties = {
+  src: string;
+  alt: string;
+  className?: string;
+  width?: string;
+  height?: string;
+};
+
+const Image: React.FC<Properties> = ({
```
Comment (dimapopovych) [inline frontend/src/libs/components/image/image.tsx:13]: If it's common ui component it should be width: 100%

File: frontend/src/packages/projects/projects-api.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
+import { type ProjectGetAllResponseDto } from 'shared/build/index.js';
```
Comment (dimapopovych) [inline frontend/src/packages/projects/projects-api.ts]:
```suggestion
import { type ProjectGetAllResponseDto } from './libs/types/types.js';
```

File: frontend/src/packages/projects/projects-api.ts [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+import { ProjectsApiPath } from 'shared/build/index.js';
```
Comment (dimapopovych) [inline frontend/src/packages/projects/projects-api.ts]: same

File: frontend/src/packages/projects/projects-api.ts [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+class ProjectsApi extends HttpApi {
+  public constructor({ baseUrl, http, storage }: Constructor) {
+    super({ path: ApiPath.PROJECTS, baseUrl, http, storage });
+  }
+
+  public async getUserProjects(): Promise<ProjectGetAllResponseDto> {
+    const response = await this.load(
+      this.getFullEndpoint(ProjectsApiPath.ROOT, {}),
```
Comment (dimapopovych) [inline frontend/src/packages/projects/projects-api.ts:22]: Your current endpoint expects /:userId as parameter, but you didn't specify it

File: backend/src/packages/projects/project.controller.ts [hunk 14]
```diff
@@ -46,6 +48,15 @@ class ProjectController extends Controller {
...
+        ),
+    });
```
Comment (andanf-e) [inline backend/src/packages/projects/project.controller.ts]: There shouldn't be a separate route for projects by userId - we already have `projects/` and within it you should fetch current user (logged in user) projects
```suggestion

```

File: backend/src/packages/projects/project.service.ts (hunk 8 above)
Comment (andanf-e) [reply backend/src/packages/projects/project.service.ts]: \+  and also this check has no sense as `items` is array and empty array is truthy value

**Review by andanf-e — DISMISSED** [review]

File: backend/src/packages/projects/libs/types/types.ts (hunk 6 above)
Comment (vbazhynov) [reply backend/src/packages/projects/libs/types/types.ts]: Dont Think so, it uses only on backend

File: frontend/src/slices/projects/projects.slice.ts (hunk 4 above)
Comment (kuslosolodka) [reply frontend/src/slices/projects/projects.slice.ts]: you can create a new dto for that, check users actions

File: backend/src/libs/interfaces/service.interface.ts [hunk 15]
```diff
@@ -1,6 +1,6 @@
 interface IService<T = unknown> {
   find(id: unknown): Promise<T>;
-  findAll(): Promise<{
+  findAll(user: unknown): Promise<{
```
Comment (what1s1ove) [inline backend/src/libs/interfaces/service.interface.ts]:
```suggestion
  findAll(): Promise<{
```
add new method `findAllByUserId`

File: backend/src/libs/interfaces/repository.interface.ts [hunk 16]
```diff
@@ -1,6 +1,6 @@
 interface IRepository<T = unknown> {
   find(id: unknown): Promise<T>;
-  findAll(): Promise<T[]>;
+  findAll(id: number): Promise<T[]>;
```
Comment (what1s1ove) [inline backend/src/libs/interfaces/repository.interface.ts]: id of what? add new method

File: frontend/src/slices/projects/projects.slice.ts [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+import { getUserProjects } from './actions.js';
+
+type State = {
+  dataStatus: ValueOf<typeof DataStatus>;
+  projects: ProjectGetAllResponseDto;
+};
+
+const initialState: State = {
+  dataStatus: DataStatus.IDLE,
+  projects: { items: [] },
```
Comment (what1s1ove) [inline frontend/src/slices/projects/projects.slice.ts]:
```suggestion
  projects: [],
```

File: frontend/src/slices/projects/projects.slice.ts (hunk 4 above)
Comment (what1s1ove) [reply frontend/src/slices/projects/projects.slice.ts]: why it resolved without any comments? @kuslosolodka is right

File: frontend/src/slices/projects/projects.slice.ts [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+
+import { DataStatus } from '~/libs/enums/enums.js';
+import { type ValueOf } from '~/libs/types/types.js';
+import { type ProjectGetAllResponseDto } from '~/packages/projects/projects.js';
+
+import { getUserProjects } from './actions.js';
+
+type State = {
+  dataStatus: ValueOf<typeof DataStatus>;
+  projects: ProjectGetAllResponseDto;
```
Comment (what1s1ove) [inline frontend/src/slices/projects/projects.slice.ts]:
```suggestion
  projects: ProjectGetAllItemResponseDto;
```

File: frontend/src/slices/projects/actions.ts [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+import { name as sliceName } from './projects.slice.js';
+
+const getUserProjects = createAsyncThunk<
+  ProjectGetAllResponseDto,
+  undefined,
+  AsyncThunkConfig
+>(`${sliceName}/get-user-projects`, (_, { extra }) => {
+  const { projectsApi } = extra;
+
+  return projectsApi.getUserProjects();
```
Comment (what1s1ove) [inline frontend/src/slices/projects/actions.ts]:
```suggestion
  return projectsApi.getProjects();
```

File: frontend/src/pages/my-projects/my-projects.tsx [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+  useAppSelector,
+  useEffect,
+} from '~/libs/hooks/hooks.js';
+import { actions as projectActions } from '~/slices/projects/projects.js';
+
+import { ProjectCard } from './components/project-card/project-card.js';
+import styles from './styles.module.scss';
+
+const MyProjects: React.FC = () => {
+  const mockUser: UserAuthResponse = {
```
Comment (what1s1ove) [inline frontend/src/pages/my-projects/my-projects.tsx]: use real user from auth slice

File: frontend/src/pages/my-projects/components/project-card/project-card.tsx [hunk 21]
```diff
@@ -0,0 +1,19 @@
+import styles from './styles.module.scss';
+
+type Properties = {
+  key: number;
```
Comment (what1s1ove) [inline frontend/src/pages/my-projects/components/project-card/project-card.tsx]: should be a full project object

File: frontend/src/packages/projects/projects-api.ts (hunk 13 above)
Comment (what1s1ove) [reply frontend/src/packages/projects/projects-api.ts:22]: should be root

File: frontend/src/packages/projects/projects-api.ts (hunk 11 above)
Comment (what1s1ove) [reply frontend/src/packages/projects/projects-api.ts]: why was it resolved without any comment @dimapopovych is right?

File: frontend/src/packages/projects/projects-api.ts (hunk 12 above)
Comment (what1s1ove) [reply frontend/src/packages/projects/projects-api.ts]: why was it resolved without any commeny @dimapopovych is right?

File: frontend/src/libs/enums/app-route.enum.ts [hunk 22]
```diff
@@ -2,6 +2,7 @@ const AppRoute = {
   ROOT: '/',
   SIGN_IN: '/sign-in',
   SIGN_UP: '/sign-up',
+  PROJECTS: '/projects',
```
Comment (what1s1ove) [inline frontend/src/libs/enums/app-route.enum.ts]:
```suggestion
  MY_PROJECTS: '/my-projects',
```

File: frontend/src/libs/components/image/image.tsx (hunk 10 above)
Comment (what1s1ove) [reply frontend/src/libs/components/image/image.tsx:13]: right

File: frontend/src/index.tsx (hunk 9 above)
Comment (what1s1ove) [reply frontend/src/index.tsx]: agreed

File: frontend/src/index.tsx [hunk 23]
```diff
@@ -17,6 +17,8 @@ import { AccountSettings } from '~/pages/account-settings/account-settings.js';
 import { Auth } from '~/pages/auth/auth.js';
 import { NotFound } from '~/pages/not-found/not-found.js';
 
+import { MyProjects } from './pages/my-projects/my-projects.js';
```
Comment (what1s1ove) [inline frontend/src/index.tsx]:
```suggestion
import { MyProjects } from '~/pages/my-projects/my-projects.js';
```

File: backend/src/packages/projects/project.service.ts [hunk 24]
```diff
@@ -15,8 +17,14 @@ class ProjectService implements Omit<IService, 'find' | 'update' | 'delete'> {
...
   }
 
-  public async findAll(): Promise<ProjectGetAllResponseDto> {
-    const items = await this.projectRepository.findAll();
+  public async findAll(
+    user: UserAuthResponse | undefined | null,
+  ): Promise<ProjectGetAllResponseDto> {
+    if (!user) {
+      throw new ApplicationError({ message: 'Unauthorized' });
+    }
```
Comment (what1s1ove) [inline backend/src/packages/projects/project.service.ts]:
```suggestion
```
this is protected route. use always shuold be on this route

File: backend/src/packages/projects/project.service.ts [hunk 25]
```diff
@@ -2,6 +2,8 @@ import { type IService } from '~/libs/interfaces/interfaces.js';
 import { ProjectEntity } from '~/packages/projects/project.entity.js';
 import { type ProjectRepository } from '~/packages/projects/project.repository.js';
 
+import { ApplicationError } from '../auth/libs/exceptions/exceptions.js';
```
Comment (what1s1ove) [inline backend/src/packages/projects/project.service.ts]:
```suggestion
import { ApplicationError } from '~/libs/exceptions/exceptions.js';
```

File: backend/src/packages/projects/project.repository.ts [hunk 26]
```diff
@@ -11,8 +11,16 @@ class ProjectRepository
     this.projectModel = projectModel;
   }
 
-  public async findAll(): Promise<ProjectEntity[]> {
-    const projects = await this.projectModel.query().execute();
+  public async findAll(id: number): Promise<ProjectEntity[]> {
```
Comment (what1s1ove) [inline backend/src/packages/projects/project.repository.ts]: id of that?  findAll should return all
```suggestion
  public async findAll(): Promise<ProjectEntity[]> {
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/packages/projects/project.repository.ts (hunk 26 above)
Comment (andanf-e) [reply backend/src/packages/projects/project.repository.ts]: for getting projects by userId I believe it should be `findByUserId` a separate method - so you don't need to change `findAll`

File: backend/src/packages/projects/project.controller.ts [hunk 27]
```diff
@@ -76,10 +77,12 @@ class ProjectController extends Controller {
...
    *                  $ref: '#/components/schemas/Project'
    */
-  private async findAll(): Promise<ApiHandlerResponse> {
+  private async findAll(
+    options: ApiHandlerOptions,
+  ): Promise<ApiHandlerResponse> {
     return {
       status: HttpCode.OK,
-      payload: await this.projectService.findAll(),
+      payload: await this.projectService.findAll(options.user),
```
Comment (andanf-e) [inline backend/src/packages/projects/project.controller.ts]:
```suggestion
      payload: await this.projectService.findByUserId(user.id),
```

**Review by andanf-e — CHANGES_REQUESTED** [review]

File: frontend/src/slices/projects/projects.slice.ts (hunk 4 above)
Comment (vbazhynov) [reply frontend/src/slices/projects/projects.slice.ts]: Got it

File: frontend/src/packages/projects/projects-api.ts (hunk 11 above)
Comment (vbazhynov) [reply frontend/src/packages/projects/projects-api.ts]: right

File: frontend/src/libs/components/header/header.tsx [hunk 28]
```diff
@@ -6,10 +6,11 @@ import { type UserAuthResponse } from '~/packages/users/users.js';
 import styles from './styles.module.scss';
 
 type Properties = {
-  user: UserAuthResponse;
+  user: UserAuthResponse | null;
```
Comment (andanf-e) [inline frontend/src/libs/components/header/header.tsx]: Because header is shown only when user is logged in
```suggestion
  user: UserAuthResponse;
```

**Review by andanf-e — CHANGES_REQUESTED** [review]

File: frontend/src/pages/my-projects/my-projects.tsx [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+    projects: state.projects.projects,
+    currentUser: state.auth.user,
+  }));
+
+  return (
+    <PageLayout>
+      <Header user={currentUser} pageName={PageName.MY_PROJECTS} />
+      <div className={styles['cards-wrapper']}>
+        {projects.map((card) => (
+          <ProjectCard key={card.id} siteName={card.name} />
```
Comment (what1s1ove) [inline frontend/src/pages/my-projects/my-projects.tsx]:
```suggestion
          <ProjectCard key={card.id} project={card} />
```

File: frontend/src/libs/enums/page-name.enum.ts [hunk 30]
```diff
@@ -0,0 +1,6 @@
+const PageName = {
```
Comment (what1s1ove) [inline frontend/src/libs/enums/page-name.enum.ts]:
```suggestion
```
do not need a enum for it. just pass a string whenever you need it

File: backend/src/packages/projects/libs/types/types.ts [hunk 31]
```diff
@@ -1,5 +1,7 @@
 export {
+  type GetProjectsResponseDto,
   type ProjectCreateRequestDto,
   type ProjectCreateResponseDto,
   type ProjectGetAllResponseDto,
+  type UserAuthResponse,
```
Comment (what1s1ove) [inline backend/src/packages/projects/libs/types/types.ts]:
```suggestion
```
it isn't a part of `project` package

File: frontend/src/packages/projects/projects-api.ts (hunk 12 above)
Comment (what1s1ove) [reply frontend/src/packages/projects/projects-api.ts]: isn't resolved

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/header/header.tsx (hunk 28 above)
Comment (vbazhynov) [reply frontend/src/libs/components/header/header.tsx]: image
method getCurrentUser return `user: UserAuthResponse | null`. In my understanding method must return only user, checking must be on startup, not in action. But this method already approved and merged,  so based on this return type in Header`UserAuthResponse | null` too, or i dont understand something?

Comment (andanf-e) [reply frontend/src/libs/components/header/header.tsx]: My projects is a protected route -> protected route requires a logged in user -> in this case you're sure that user is not null

Comment (vbazhynov) [reply frontend/src/libs/components/header/header.tsx]: I understand it, but in store user with type `UserAuthResponse | null`, so if I leave in Header component only `UserAuthResponse` there is an error, how can i resolve it? i'm confused

Comment (andanf-e) [reply frontend/src/libs/components/header/header.tsx]: just cast to `UserAuthResponse` in place where you need it

File: frontend/src/libs/components/header/header.tsx [hunk 32]
```diff
@@ -2,15 +2,16 @@ import avatarImage from '~/assets/img/default-avatar-profile-icon.svg';
...
 import { AppRoute } from '~/libs/enums/app-route.enum';
 import { useCallback } from '~/libs/hooks/hooks.js';
-import { type UserAuthResponse } from '~/packages/users/users.js';
+import { type ProjectUserEssence } from '~/packages/projects/projects.js';
 
 import styles from './styles.module.scss';
 
 type Properties = {
-  user: UserAuthResponse;
+  user: ProjectUserEssence | null;
```
Comment (andanf-e) [inline frontend/src/libs/components/header/header.tsx]: For what reason common header should know anything about any Project entity?
```suggestion
  user: UserAuthResponse;
```

**Review by andanf-e — CHANGES_REQUESTED** [review]

File: backend/src/packages/projects/libs/types/types.ts (hunk 31 above)
Comment (what1s1ove) [reply backend/src/packages/projects/libs/types/types.ts]: isn't resolved

File: shared/src/packages/projects/libs/types/project-user-essence.type.ts [hunk 33]
```diff
@@ -0,0 +1,9 @@
+type ProjectUserEssence = {
```
Comment (what1s1ove) [inline shared/src/packages/projects/libs/types/project-user-essence.type.ts]: you don't need this type. remove it

File: frontend/src/libs/components/header/header.tsx (hunk 32 above)
Comment (what1s1ove) [reply frontend/src/libs/components/header/header.tsx]: isn't resolved

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: shared/src/packages/projects/libs/types/get-projects-response-dto.type.ts [hunk 34]
```diff
@@ -0,0 +1,7 @@
+import { type GetProjectsItemResponseDto } from './get-projects-item-response-dto.type.js';
+
+type GetProjectsResponseDto = {
```
Comment (what1s1ove) [inline shared/src/packages/projects/libs/types/get-projects-response-dto.type.ts]:
```suggestion
type ProjectGetAllResponseDto = {
```
check how it looks in other packages (ex. users)

File: shared/src/packages/projects/libs/types/get-projects-item-response-dto.type.ts [hunk 35]
```diff
@@ -0,0 +1,7 @@
+type GetProjectsItemResponseDto = {
```
Comment (what1s1ove) [inline shared/src/packages/projects/libs/types/get-projects-item-response-dto.type.ts]:
```suggestion
type ProjectGetAllItemResponseDto = {
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/pages/my-projects/my-projects.tsx [hunk 36]
```diff
@@ -0,0 +1,56 @@
...
+  const { projects, currentUser } = useAppSelector((state) => ({
+    projects: state.projects.projects,
+    currentUser: state.auth.user as UserAuthResponse,
+  }));
+
+  const hasProjects = projects.length > 0;
+
+  return (
+    <PageLayout>
+      <Header user={currentUser} pageName="My Projects" />
```
Comment (Alyona-Pisotska) [inline frontend/src/pages/my-projects/my-projects.tsx]: You don`t need Header there. PageLayout has Header

**Review by Alyona-Pisotska — CHANGES_REQUESTED** [review]

File: frontend/src/libs/enums/page-name.enum.ts (hunk 30 above)
Comment (what1s1ove) [reply frontend/src/libs/enums/page-name.enum.ts]: isn't resolved

File: frontend/src/packages/sites/sites-api.ts [hunk 37]
```diff
@@ -0,0 +1,41 @@
...
+  public constructor({ baseUrl, http, storage }: Constructor) {
+    super({ path: '', baseUrl, http, storage });
+  }
+
+  public async getByProjectId(
+    projectId: number,
+  ): Promise<SiteGetAllResponseDto> {
+    const response = await this.load(
+      this.getFullEndpoint(
+        configureString(ApiPath.PROJECTS_$PROJECT_ID_SITES, SitesApiPath.ROOT, {
```
Comment (what1s1ove) [inline frontend/src/packages/sites/sites-api.ts]:
```suggestion
        configureString(ApiPath.SITES, ProjectsApiPath.PROJECT_PROJECT_ID, {
```
the final path should be `/sites/project/:projectId`

File: shared/src/libs/enums/api-path.enum.ts [hunk 38]
```diff
@@ -1,7 +1,7 @@
 const ApiPath = {
   USERS: '/users',
   AUTH: '/auth',
-  SITES: '/sites',
+  PROJECTS_$PROJECT_ID_SITES: '/projects/:projectId/sites',
```
Comment (what1s1ove) [inline shared/src/libs/enums/api-path.enum.ts]:
```suggestion
```
should be a part of nested pacakge

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/packages/sites/sites-api.ts (hunk 37 above)
Comment (what1s1ove) [reply frontend/src/packages/sites/sites-api.ts]: isn't resolved

File: shared/src/libs/enums/api-path.enum.ts (hunk 38 above)
Comment (what1s1ove) [reply shared/src/libs/enums/api-path.enum.ts]: isn't resolved

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/slices/projects/actions.ts [hunk 39]
```diff
@@ -0,0 +1,18 @@
...
+import { type AsyncThunkConfig } from '~/libs/types/async-thunk-config.type.js';
+import { type ProjectGetAllResponseDto } from '~/packages/projects/projects.js';
+
+import { name as sliceName } from './projects.slice.js';
+
+const getUserProjects = createAsyncThunk<
+  ProjectGetAllResponseDto,
+  undefined,
+  AsyncThunkConfig
+>(`${sliceName}/get-projects`, (_, { extra }) => {
```
Comment (jaroslawtt) [inline frontend/src/slices/projects/actions.ts]:
```suggestion
>(`${sliceName}/get-projects`, async (_, { extra }) => {
```

File: frontend/src/slices/projects/actions.ts [hunk 40]
```diff
@@ -0,0 +1,18 @@
...
+  const { projectsApi } = extra;
+
+  return projectsApi.getProjects();
```
Comment (jaroslawtt) [inline frontend/src/slices/projects/actions.ts]:
```suggestion
  return await projectsApi.getProjects();
```

File: frontend/src/slices/projects/actions.ts (hunk 39 above)
Comment (vbazhynov) [reply frontend/src/slices/projects/actions.ts]: +

File: shared/src/packages/projects/libs/enums/projects-api-path.enum.ts [hunk 41]
```diff
@@ -1,5 +1,6 @@
 const ProjectsApiPath = {
   ROOT: '/',
+  PROJECT_PROJECT_ID: '/project/:projectId',
```
Comment (what1s1ove) [inline shared/src/packages/projects/libs/enums/projects-api-path.enum.ts]:
```suggestion
```
it part of the `sites` api path

File: frontend/src/libs/enums/app-route.enum.ts [hunk 42]
```diff
@@ -2,6 +2,8 @@ const AppRoute = {
   ROOT: '/',
   SIGN_IN: '/sign-in',
   SIGN_UP: '/sign-up',
+  MY_PROJECTS: '/my-projects',
+  SITES_PROJECTS_$PROJECT_ID: '/sites/project/:projectId',
```
Comment (what1s1ove) [inline frontend/src/libs/enums/app-route.enum.ts]:
```suggestion
 PROJECTS_$PROJECT_ID_SITES: '/projects/:projectId/sites'
```

File: backend/src/packages/sites/site.controller.ts [hunk 43]
```diff
@@ -104,16 +119,18 @@ class SiteController extends Controller {
...
       status: HttpCode.OK,
-      payload: await this.siteService.findAll(),
+      payload: await this.siteService.findByProjectId(options.params.projectId),
     };
   }
 
   /**
    * @swagger
-   * /sites:
+   * /projects/:projectId/sites:
```
Comment (what1s1ove) [inline backend/src/packages/sites/site.controller.ts]:
```suggestion
   * /project/:projectId/sites:
```

File: backend/src/packages/sites/site.controller.ts [hunk 44]
```diff
@@ -87,9 +102,9 @@ class SiteController extends Controller {
...
```
Comment (what1s1ove) [inline backend/src/packages/sites/site.controller.ts]:
```suggestion
   * /project/:projectId/sites:
```

File: backend/src/packages/sites/site.controller.ts [hunk 45]
```diff
@@ -104,16 +119,18 @@ class SiteController extends Controller {
    *                     $ref: '#/components/schemas/Site'
    *                   minItems: 0
    */
-  private async findAll(): Promise<ApiHandlerResponse> {
+  private async findByProjectId(
```
Comment (what1s1ove) [inline backend/src/packages/sites/site.controller.ts]:
```suggestion
  private async findAllByProjectId(
```

File: backend/src/packages/sites/site.controller.ts [hunk 46]
```diff
@@ -104,16 +119,18 @@ class SiteController extends Controller {
...
+    options: ApiHandlerOptions<{ params: SiteGetByProjectParametersDto }>,
+  ): Promise<ApiHandlerResponse> {
     return {
       status: HttpCode.OK,
-      payload: await this.siteService.findAll(),
+      payload: await this.siteService.findByProjectId(options.params.projectId),
```
Comment (what1s1ove) [inline backend/src/packages/sites/site.controller.ts]:
```suggestion
      payload: await this.siteService.findAllByProjectId(options.params.projectId),
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: shared/src/packages/sites/libs/enums/sites-api-path.enum.ts [hunk 47]
```diff
@@ -1,6 +1,7 @@
 const SitesApiPath = {
   ROOT: '/',
   SECTIONS_BY_SITE: '/:siteId/sections',
+  PROJECT_PROJECT_ID: '/project/:projectId',
```
Comment (what1s1ove) [inline shared/src/packages/sites/libs/enums/sites-api-path.enum.ts]:
```suggestion
  PROJECT_$PROJECT_ID: '/project/:projectId',
```
all dynamic parameters should start with `$`

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**